gapic/src/main/com/google/gapid/views/TracerDialog.java

```diff
@@ -438,6 +457,28 @@ protected void configureDialog(DirectoryDialog dialog) {
           }
         });
 
+        Listener filledListener = e -> {
+            requiredFieldMessage.setVisible(!this.isReady());
+            deviceLabel.setForeground(getDisplay().getSystemColor(getSelectedDevice() == null ? SWT.COLOR_RED : SWT.COLOR_BLACK));
+            directoryLabel.setForeground(getDisplay().getSystemColor(directory.getText().isEmpty() ? SWT.COLOR_RED : SWT.COLOR_BLACK));
+            fileLabel.setForeground(getDisplay().getSystemColor(file.getText().isEmpty() ? SWT.COLOR_RED : SWT.COLOR_BLACK));
+
+            TraceTypeCapabilities config = getSelectedApi();
+            if(config != null) {
+              apiLabel.setForeground(getDisplay().getSystemColor(SWT.COLOR_BLACK));
+              targetLabel.setForeground(getDisplay().getSystemColor(
+                  (config.getRequiresApplication() && traceTarget.getText().isEmpty()) ?
+                      SWT.COLOR_RED : SWT.COLOR_BLACK));
+            } else {
+              apiLabel.setForeground(getDisplay().getSystemColor(SWT.COLOR_RED));
+              targetLabel.setForeground(getDisplay().getSystemColor(SWT.COLOR_BLACK));
```

Comment:
LGTM. Nit is that getDisplay().getSystemColor(SWT.COLOR_RED) could be pre-saved for code conciseness.

Reply:
And btw most of the constant color values are defined in gapic/src/main/com/google/gapid/widgets/Theme.java. It would be a more consistent pattern to our project if defining the color there and referring to it.